### PR TITLE
🚀 Expose `scrollControlDisabledMaxHeightRatio` to the modal bottom sheet

### LINKED ISSUE #129690: [ModalBottomSheet] Allow configuring the max height ratio when `isScrollControlled` is false
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

There is a default value when `isScrollControlled` is false for sheets:
https://github.com/flutter/flutter/blob/14c04cd0f6a0a62b455bda958574e13482234281/packages/flutter/lib/src/material/bottom_sheet.dart#L598-L600
The value is hard-coded and not allowed to be configured. In order to apply custom constraints in the scenario, users must wrap a `ConstrainedBox`.

### Proposal

Expose the value as a parameter `scrollControlDisabledMaxHeightRatio` for:
- `_BottomSheetLayoutWithSizeListener`
- `_RenderBottomSheetLayoutWithSizeListener`
- `_ModalBottomSheet`
- `ModalBottomSheetRoute`
- `showModalBottomSheet`

### PR DESCRIPTION
**Pull request by AlexV525**
Adding the `scrollControlDisabledMaxHeightRatio` parameter for modal bottom sheet widgets, and using the default value `9.0 / 16.0` to avoid breaking.

Resolves #129690.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/bottom_sheet.dart [hunk 1]
```diff
@@ -591,13 +608,13 @@ class _RenderBottomSheetLayoutWithSizeListener extends RenderShiftedBox {
     return _getSize(constraints);
   }
 
-    BoxConstraints _getConstraintsForChild(BoxConstraints constraints) {
+  BoxConstraints _getConstraintsForChild(BoxConstraints constraints) {
```
Comment (Renzo-Olivares) [inline packages/flutter/lib/src/material/bottom_sheet.dart:611]: Nice catch!

File: packages/flutter/lib/src/material/bottom_sheet.dart [hunk 2]
```diff
@@ -26,6 +26,7 @@ const Duration _bottomSheetExitDuration = Duration(milliseconds: 200);
 const Curve _modalBottomSheetCurve = decelerateEasing;
 const double _minFlingVelocity = 700.0;
 const double _closeProgressThreshold = 0.5;
+const double _scrollControlledMaxHeightRatio = 9.0 / 16.0;
```
Comment (Renzo-Olivares) [inline packages/flutter/lib/src/material/bottom_sheet.dart:29]: nit: maybe `_defaultScrollControlledMaxHeightRatio`.

File: packages/flutter/lib/src/material/bottom_sheet.dart [hunk 3]
```diff
@@ -842,6 +863,12 @@ class ModalBottomSheetRoute<T> extends PopupRoute<T> {
   /// to have the bottom sheet be draggable.
   final bool isScrollControlled;
 
+  /// The max height constraint ratio for the bottom sheet
+  /// when [isScrollControlled] set to true.
+  ///
```
Comment (Renzo-Olivares) [inline packages/flutter/lib/src/material/bottom_sheet.dart:868]: Nit: maybe note that when [isScrollControlled] is set to false there is no constraint ratio.

**Review by Renzo-Olivares — APPROVED** [review]
LGTM with some small nits. Thank you for your contribution!

**Comment by AlexV525** [comment]
@Renzo-Olivares PTAL

**Comment by Renzo-Olivares** [comment]
@AlexV525 looks like some tests are failing, can you take a look?

**Comment by AlexV525** [comment]
> @AlexV525 looks like some tests are failing, can you take a look?

Yeah I'll figure them out as soon as I'm available.

**Comment by AlexV525** [comment]
@Renzo-Olivares Should be good to go.

File: packages/flutter/lib/src/material/bottom_sheet.dart [hunk 4]
```diff
@@ -842,6 +863,13 @@ class ModalBottomSheetRoute<T> extends PopupRoute<T> {
   /// to have the bottom sheet be draggable.
   final bool isScrollControlled;
 
+  /// The max height constraint ratio for the bottom sheet
+  /// when [isScrollControlled] set to true,
```
[the remark this answers is not in the document]
Comment (AlexV525) [reply packages/flutter/lib/src/material/bottom_sheet.dart:867]: Huge thanks for the catch!

**Comment by AlexV525** [comment]
@Renzo-Olivares I've updated these names to `scrollControlDisabledMaxHeightRatio` series.

**Review by Renzo-Olivares — APPROVED** [review]
LGTM

**Pull request merged**